### ISSUE TITLE
crafts-flake FTBFS against NixOS 23.11

### ISSUE THREAD

**Issue by flexiondotorg** [jnsgruk/crafts-flake#41]
crafts-flake 0.5.x FTBFS on my NixOS 23.11 systems. I've pinned the flake to 0.4.3 to workaround the issue.

Here's the error:

```
error: builder for '/nix/store/bsw5x8h2ryf2wh4db04763qv33gx32n2-python3.11-pydantic-1.10.12.drv' failed with exit code 2;
       last 10 log lines:
       >   File "/build/source/docs/build/main.py", line 31, in <module>
       >     sys.exit(main())
       >              ^^^^^^
       >   File "/build/source/docs/build/main.py", line 24, in main
       >     from exec_examples import exec_examples
       >   File "/build/source/docs/build/exec_examples.py", line 17, in <module>
       >     from devtools import PrettyFormat
       > ModuleNotFoundError: No module named 'devtools'
       > make: *** [Makefile:118: docs] Error 1
       > /nix/store/kv5wkk7xgc8paw9azshzlmxraffqcg0i-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/bsw5x8h2ryf2wh4db04763qv33gx32n2-python3.11-pydantic-1.10.12.drv'.
error: 1 dependencies of derivation '/nix/store/d0ggrr7grgswkbpr89d16l3ks8w76xad-python3.11-craft-archives-1.1.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7yar9pdm2f9r0mp704sx8wdk6g5y2d0d-python3.11-fastapi-0.103.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rmrl7dg66dsq8ws86791s8b6f3cbdfks-python3.11-inflect-7.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/45gvnm1vy0s72k7ic5j9rwy2s6fm93zb-python3.11-versioningit-2.2.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/q83j834v2naijcqhaq9fjs18789m2r3m-snapcraft-8.0.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lb146l6s5w6h9dszcf17k609q7ngcna4-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xc9cqv70fj5ddjv4k8qlr6yp9fwk6mnc-snapcraft-8.0.1_fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sd9n70immxnxchvbbiwvcgfbr3khcf94-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qqmj20ilin9nl07bg43ychzbygd9x6gi-nixos-system-ripper-23.11.20240124.a77ab16.drv' failed to build
```

**Comment by jnsgruk** [comment]
Hmm, that's a bit strange. I can't repro this, and I also can't really understand why it's trying to build `pydantic` `1.10.12`. I'm overlaying pydantic, but it's pinned to `1.10.13`.

Are you able to build locally with `nix build .#snapcraft`?

**Comment by jnsgruk** [comment]
Aha - I found [this](https://github.com/wimpysworld/nix-config/blob/733fd2fcc28fcdf11f62b7b40c55eb32b0ca6a08/flake.nix#L31C5-L31C53). Can you try removing the `follows`, or have it follow `nixpkgs-unstable`?

**Comment by jnsgruk** [comment]
Speaking out of band, the build failures seem to correlate with the test suites of `snapcraft`, and only on low core count systems. v0.6.0 disabled the unit test suites for snapcraft, as the failures were in parts of nested dependencies that aren't actually used. Will pick up fixing the tests another time.

**Comment by jnsgruk** [comment]
Now fixed in 0.6.1
